modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -133,20 +189,152 @@ const MenuList = styled('ul')<Pick<SelectProps, 'error' | 'theme'>>(
   })
 );
 
+const generatePopperOptions = (
+  props: Pick<SelectMenuProps, 'isAutoFlipped' | 'isAutoFocused' | 'isFlipped' | 'menuRef'>
+) => {
+  const {isAutoFlipped, isAutoFocused, isFlipped, menuRef} = props;
+
+  let fallbackPlacements: Placement[] = [];
+  if (isAutoFlipped) {
+    fallbackPlacements = isFlipped ? ['bottom-start'] : ['top-start'];
+  }
+
+  const modifiers = [
+    {
+      name: 'flip',
+      options: {
+        fallbackPlacements,
+      },
+    },
+    {
+      name: 'offset',
+      options: {
+        offset: ({
+          placement,
+          reference,
+          popper,
+        }: {
+          placement: Placement;
+          reference: Rect;
```

Comment:
Is `Rect` necessary if you inline these modifiers? I don't see any benefit extracting out into a variable if inlining can drop extra TS burden. The config is already called `modifiers`, so we're not loosing the abstraction of a name

Reply:
Do you mean having `generatePopperOptions` return like so:
```
return {
  modifiers: [
    {
      name: 'offset',
      options: {
        offset: ({reference}) => {
...
```
Or even inlining `modifiers` within the `popperOptions` prop:
```
<Popper
  popperOptions={{
    modifiers: [
      {
        name: 'offset',
        options: {
          offset: ({reference}) => {
...
```
In both cases, TS complains about `reference` implicitly having an `any` type.